pkg/networkmanager/firewall-client.js

```diff
@@ -50,6 +51,15 @@ function initFirewalldDbus() {
             return;
         }
 
+        /* As certain dbus signal callbacks might change the firewall frequently
+         * in a short period of time, prevent rapid succession of renders by
+         * debouncing the ('changed') event */
+        firewall.debouncedEvent = debounce(300, event => firewall.dispatchEvent(event));
```

Comment:
The debouncing could also be split out into a separate commit/PR. (Not a strong opinion, just a random idea)